gspread/utils.py

```diff
@@ -979,6 +985,116 @@ def to_records(
     return [dict(zip(headers, row)) for row in values]
 
 
+def _expand_right(values: List[List[str]], start: int, end: int, row: int) -> int:
+    """This is a private function, returning the column index of the first empty cell
+    on the given row.
+
+    Search starts from ``start`` index column.
+    Search ends on ``end`` index column.
+    Searches only in the row pointed by ``row``.
+
+    If no empty value is found, it will return the given ``end`` index.
+    """
+    for column in range(start, end):
+        if values[row][column] == "":
+            return column
+
+    return end
+
+
+def _expand_bottom(values: List[List[str]], start: int, end: int, col: int) -> int:
+    """This is a private function, returning the row index of the first empty cell
+    on the given column.
+
+    Search starts from ``start`` index row.
+    Search ends on ``end`` index row.
+    Searches only in the column pointed by ``col``.
+
+    If no empty value is found, it will return the given ``end`` index.
+    """
+    for rows in range(start, end):
+        if values[rows][col] == "":
+            return rows
+
+    return end
+
+
+def find_table(
+    values: List[List[str]],
+    start_range: str,
+    direction: TableDirection = TableDirection.table,
+) -> List[List[str]]:
+    """Expands a list of values based on non-null adjacent cells.
+
+    Expand can be done in 3 directions defined in :class:`~gspread.utils.TableDirection`
+
+        * ``TableDirection.right``: expands only same row as starting cell on the right side up to
+          first null/empty cell
+        * ``TableDirection.down``: expands only same column as starting cell to the bottom up to
+          first null/empty cell
+        * ``TableDirection.table``: expands in both direction, first right then down.
```

Comment:
"expands right until the first empty cell, then down until the first empty cell"